src/main.cc

```diff
@@ -4,40 +4,63 @@
 namespace {
 
 NAN_METHOD(SetPassword) {
+  Nan::Utf8String serviceNan(info[0]);
+  std::string service(*serviceNan, serviceNan.length());
```

Comment:
`nan` has it's own [UTF8 string abstraction](https://github.com/nodejs/nan/blob/master/doc/v8_misc.md#api_nan_utf8_string) that works across all the versions of v8, instead of `#ifdef`-ing across all the different `Utf8Value` versions that exist across different V8 versions, and this was the simplest way I could find to convert it into an array of bytes to then pass through to the platform-specific code.

If anyone knows a more succinct way to achieve this please let me know.